datafusion/common/src/scalar/mod.rs

```diff
@@ -2225,6 +2225,34 @@ impl ScalarValue {
         }
     }
 
+    pub fn try_from_array_ignore_nulls_first(
+        array: &dyn Array,
+        mut index: usize,
+        ignore_nulls: bool,
+    ) -> Result<Self> {
+        // If ignoring nulls, find the next non-null index.
+        if ignore_nulls {
+            while index < (array.len() - 1) && !array.is_valid(index) {
```

Comment:
it is probably cheaper to get all null indices first? and then find index in the array?
```
array.nulls().unwrap().valid_indices().collect::<Vec<_>>()
```